cmd/radio.go

```diff
@@ -185,18 +192,55 @@ type erasureConfig struct {
 	clnts  []bucketClient
 }
 
+func clientID(cfg remoteConfig) string {
+	hash := sha256.New()
+	bytes, err := msgpack.Marshal(&cfg)
+	if err != nil {
+		return fmt.Sprintf("%s%s", cfg.Bucket, cfg.Endpoint)
+	}
+	hash.Write(bytes)
+	hashBytes := hash.Sum(nil)
+	return hex.EncodeToString(hashBytes)
+}
+
+const healthCheckInterval = time.Second * 5
+
 func newBucketClients(bcfgs []remoteConfig) ([]bucketClient, error) {
 	var clnts []bucketClient
 	for _, bCfg := range bcfgs {
 		clnt, err := newS3(bCfg.Bucket, bCfg.Endpoint, bCfg.AccessKey, bCfg.SecretKey, bCfg.SessionToken)
 		if err != nil {
 			return nil, err
 		}
+
 		clnts = append(clnts, bucketClient{
 			Core:   clnt,
 			Bucket: bCfg.Bucket,
+			ID:     clientID(bCfg),
 		})
 	}
+	go func() {
+		for {
+			g := errgroup.WithNErrs(len(clnts))
+			for index := range clnts {
+				index := index
+				g.Go(func() error {
+					var perr error
+					_, perr = clnts[index].BucketExists(clnts[index].Bucket)
```

Comment:
perhaps we can use the endpoint to automatically do liveness/readiness checks instead? which are lighter on the S3 API side? and of course anonymous in nature.

Reply:
BTW lets convert this into a function and call it as `go healthCheck()` and make this as part of radioObjects() implementation on how to implement a healthcheck. 

Otherwise the access to `clnts` is not global and updating pointers with fields .Up leads to races during unexpected access..